packages/ui/src/components/va-form/VaForm.vue

```diff
@@ -8,115 +9,120 @@
 </template>
 
 <script lang="ts">
-import { inject, provide } from 'vue'
-import { Options, mixins, prop, Vue, setup } from 'vue-class-component'
-import { FormProvider, FormServiceKey, FormChild } from './consts'
+import {
+  defineComponent,
+  PropType,
+  ref,
+  Ref,
+  onMounted,
+  onUnmounted,
+  nextTick,
+  provide,
+  inject,
+} from 'vue'
+
+import { FormServiceKey, FormChild } from './consts'
+
+const isVaForm = (value: any) => !!value.focusInvalid
+
+export default defineComponent({
+  name: 'VaForm',
+  emits: ['validation'],
+  props: {
+    autofocus: { type: Boolean as PropType<boolean>, default: false },
+    tag: { type: String as PropType<string>, default: 'div' },
+  },
 
-class FormProps {
-  autofocus = prop<boolean>({ type: Boolean, default: false })
-  tag = prop<string>({ type: String, default: 'div' })
-}
+  setup (props, { emit }) {
+    const nestedFormElements: Ref<FormChild[]> = ref([])
 
-const FormPropsMixin = Vue.with(FormProps)
+    const parentFormProvider = () => ({ ...inject(FormServiceKey, undefined) })
```

Comment:
Can we do this?:
`const parentFormProvider = inject(FormServiceKey, undefined)`